### PR TITLE
Create Transaction Verifier and switch runtime/api to use it

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Adding TX Verifier that mimics current Runtime checks.
It also verifies AccessKeys, but they are not tested and you can't add one yet.
Remove `&mut` from `get` resulted in a lot of files touched.
Also rust_fmt modified some style in the touched files 
For #687

### CODE REVIEW

File: runtime/primitives/src/account.rs [hunk 1]
```diff
@@ -18,3 +18,18 @@ impl Account {
         Account { public_keys, nonce: 0, amount, staked: 0, code_hash }
     }
 }
+
+/// Limited Access key for some account. It implies to properties: account_id of the owner and
```
Comment (bowenwang1996) [inline runtime/primitives/src/account.rs]: two properties

Comment (bowenwang1996) [reply runtime/primitives/src/account.rs]: Also implies sounds weird

File: runtime/runtime/src/lib.rs [hunk 2]
```diff
@@ -163,100 +164,63 @@ impl Runtime {
...
-                    TransactionBody::AddKey(ref t) => system::add_key(state_update, t, &mut sender),
-                    TransactionBody::DeleteKey(ref t) => {
-                        system::delete_key(state_update, t, &mut sender)
-                    }
-                }
+        let VerificationData { originator_id, mut originator, .. } = {
+            let verifier = TransactionVerifier::new(state_update);
+            verifier.verify_transaction(transaction)?
+        };
+        // Update nonce regardless of whether the transaction succeeds. This is done
```
Comment (bowenwang1996) [inline runtime/runtime/src/lib.rs]: This comment needs to be changed

File: runtime/primitives/src/account.rs (hunk 1 above)
Comment (evgenykuzyakov) [reply runtime/primitives/src/account.rs]: Updated comments

File: runtime/runtime/src/lib.rs (hunk 2 above)
Comment (evgenykuzyakov) [reply runtime/runtime/src/lib.rs]: Good point

**Comment by evgenykuzyakov** [comment]
Run nearlib tests locally. It passes all tests

**Pull request merged**